src/core/mac/mac.cpp

```diff
@@ -237,17 +260,8 @@ otError Mac::ConvertBeaconToActiveScanResult(Frame *aBeaconFrame, otActiveScanRe
 
 otError Mac::UpdateScanChannel(void)
 {
-    otError error = OT_ERROR_NONE;
-
-    do
-    {
-        mScanChannel++;
-        VerifyOrExit(mScanChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_NOT_FOUND);
-
-    } while ((mScanChannels & (1U << mScanChannel)) == 0);
-
-exit:
-    return error;
+    return (mScanChannel == 0) ? mScanChannelMask.GetFirstChannel(mScanChannel)
```

Comment:
What if `OT_RADIO_CHANNEL_MIN` is 0?  Again, I know this is not true for IEEE 802.15.4-2006 2.4 GHz.

Reply:
I feel the correct way for this would be to have a boolean variable to indicate if this is first channel or next. I was hoping to avoid the extra variable by assigning a special value to `mScanChannel` when we start scan.  Can possibly use `0xff` instead.

Reply:
Actually I think better way would be to change the iterator methods and have only `GetNextChannel(channel)` and special channel value defined by the `ChannelMask` itself to pas in when we want to get the first channel.